tests/chainer_tests/functions_tests/array_tests/test_resize_images.py

```diff
@@ -0,0 +1,94 @@
+import unittest
+
+import numpy
+
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+@testing.parameterize(*testing.product({
+    'in_shape': [(2, 3, 8, 6), (2, 1, 4, 6)],
+}))
+class TestResizeImagesForwardIdentity(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(
+            size=self.in_shape).astype(numpy.float32)
+
+    def check_forward(self, x, output_shape):
+        y = functions.resize_images(x, output_shape)
+        testing.assert_allclose(y.data, x)
+
+    def test_forward_cpu(self):
+        self.check_forward(self.x, output_shape=self.in_shape[2:])
+
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x), output_shape=self.in_shape[2:])
+
+
+class TestResizeImagesForwardScale(unittest.TestCase):
```

Comment:
Ditto. It seems that your proposed function also accepts `float64` and `float16`.